### ISSUE TITLE
unreliable cluster creation and template update

### ISSUE THREAD

**Issue by paulklassen** [vmware/container-service-extension#258]
CSE 1.2.7
VCD-CLI 21.0.2
vCloud Director 9.1 (on vCenter 6.5 and ESXi 6.5)
Python 2.7.5

So I have actually two problems who may or may not have the same common root.

1. When trying to update the templates with
`cse install -c /path/to/config.yaml --template photon-v2 --update --ext skip`
it does not work most of the times and if it does, it takes a very long time, like 3-4 hours.
I noticed that it either fails during importing the ova file or while cloning the *-temp VM template.
And I can't find any hints in the logs for that behaviour

2. Creating a cluster takes also very long (6+ hours) and usually it fails after creating the master node. There are no errors or anything, but nothing is happening at this point either in vCenter nor in vCloud Director, though the "Creating master node" process keeps on running.
"vcd cse cluster list" shows me the cluster name, but no IP in "IP master" and also only 1 VM (the master node), but no worker nodes.

**Comment by sakthisunda** [comment]
1. Please upload the logs under the following paths, when the actual failure happened.
   <CSE_INSTALL_LOCATION>/cse-logs/*.log ( mainly cse-server-debug.log)
   <CSE_INSTALL_LOCATION>/vcd.log
2. If you can also cut and paste what you see on the console, when you 
    a) update the templates b) create cluster 
3. Finally, you mentioned it takes longer time to import ova, create cluster. Any time, you got a successful upgrade and cluster creation ?

This will be useful for us to investigate. thanks

**Comment by paulklassen** [comment]
Hi @sakthisunda i will create a blank new logfile for further investigation.
For that I will try to create a new Cluster, when it's not ready until 30 Minutes, I will cancel the cluster creation and post the log content here.

How long does the Cluster Creation usually takes in a normal working environment?
I think this shouldn't take longer than a couple of minutes?

**Comment by sakthisunda** [comment]
In a normal working environment, everything including template download and ova import and cluster creation take 20 mts or less, depending on internet speed

**Comment by paulklassen** [comment]
Hi! So basically I got CSE up and running again...

I think we had a wrong Setup. We had two vCloud Director Server which only shared a vCloud database (on another Server), but both had their very own local RabbitMQ server and CSE server running and in use. 
I did some research and found some Informations, that in a multi-cell vCloud Director Setup the vcloud cells must share not only the database server but also RabbitMQ server and CSE server which all should be running on different servers (or for high availability in separated RabbitMQ-Cluster, CSE-Server Cluster and Postgres-Server-Cluster with some kind of Load Balancer in front of each cluster).
